scripts/bootstrap/install.py

```diff
@@ -173,8 +184,21 @@ def sha256_file(path: Path):
 
     print(f"Installed executables for python{version}")
 
-    # Cleanup
-    install_dir.with_suffix(".tmp").rmdir()
-    (THIS_DIR / filename).unlink()
 
-print("Done!")
+if __name__ == "__main__":
+    if INSTALL_DIR.exists():
+        print("Removing existing installations...")
+        shutil.rmtree(INSTALL_DIR)
+
+    # Download in parallel
+    with concurrent.futures.ProcessPoolExecutor(max_workers=len(versions)) as executor:
```

Comment:
```suggestion
    with concurrent.futures.ProcessPoolExecutor(max_workers=min(os.cpu_count(), len(versions))) as executor:
```

Might be worth to limit by what's available

Reply:
I don't think overallocating here has much of an affect, I'm just going to leave this as-is for now I hope to move this all to Rust in the near future.